services/language_translation/v1.js

```diff
@@ -13,55 +13,208 @@
+      version: 'v2'
+    });
+    language_translation.getIdentifiableLanguages(null,
+      function (err, languages) {
+        if (err) {
+          res.json(err);
+        }
+        else {
+          res.json(languages);
+        }
+      });
   });
 
   function SMTNode(config) {
     RED.nodes.createNode(this, config);
     var node = this;
 
-    this.on('input', function(msg) {
+    this.doTranslate = function (msg, model_id) {
+      language_translation = watson.language_translation({
+        username: username,
+        password: password,
+        version: 'v2'
+      });
+
+      node.status({
+        fill: 'blue',
+        shape: 'dot',
+        text: 'requesting'
+      });
+      language_translation.translate({
+        text: msg.payload,
+        model_id: model_id
+      },
+      function (err, response) {
+        node.status({})
+        if (err) {
+          node.error(err, msg);
+        } else {
+          msg.payload = response.translations[0].translation;
+        }
+        node.send(msg);
+      });
+    };
+
+    this.doTrain = function (msg, model_id, filetype) {
+      language_translation = watson.language_translation({
+        username: username,
+        password: password,
+        version: 'v2'
+      });
+
+      node.status({
+        fill: 'blue',
+        shape: 'dot',
+        text: 'requesting training'
+      });
+
+      temp.open({
+        suffix: '.xml'
+      }, function (err, info) {
+        if (!err) {
+          fs.write(info.fd, msg.payload);
+          var params = {};
+          
+          switch (filetype) {
+          case 'forcedglossary':
+            params = {
+              name: msg.filename.replace(/[^0-9a-z]/gi, ''),
+              base_model_id: 'en-es',
+              forced_glossary: fs.createReadStream(info.path)
+            };
+            break;
+          case 'parallelcorpus':
+            params = {
+              name: msg.filename.replace(/[^0-9a-z]/gi, ''),
+              base_model_id: 'en-es',
+              parallel_corpus: fs.createReadStream(info.path)
+            };
+            break;
+          case 'monolingualcorpus':
+            params = {
+              name: msg.filename.replace(/[^0-9a-z]/gi, ''),
+              base_model_id: 'en-es',
+              monolingual_corpus: fs.createReadStream(info.path)
+            };
+            break;
+          }
+          // Watson SDK call
+          language_translation.createModel(params,
+            function (err, model) {
+              node.status({});
+              if (err) {
+                node.status({
+                  fill: 'red',
+                  shape: 'ring',
+                  text: 'call to translation service failed'
+                });
+                node.error(err, msg);
+              } else {
+                node.status({
+                  fill: 'green',
+                  shape: 'dot',
+                  text: 'model sent to training'
+                });
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Line is too long.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2398740744/issues/source?bid=3225115&fileBranchId=3231190#l164)